From 3cfaec48b6f33f98f0ca201533b1faee70f111d1 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 9 Sep 2019 14:20:42 +0700
Subject: [PATCH 251/328] arm64: dts: renesas: r8a774b1: Add Audio-DMAC support

This patch adds Audio-DMAC[0,1] support for the R8A774B1 (RZ/G2N) SoC.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774b1.dtsi | 62 ++++++++++++++++++++++++++++++-
 1 file changed, 60 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774b1.dtsi b/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
index 9101dbd8..2a116b5 100644
--- a/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
@@ -1499,13 +1499,71 @@
 		};
 
 		audma0: dma-controller@ec700000 {
+			compatible =	"renesas,dmac-r8a774b1",
+					"renesas,rcar-dmac";
 			reg = <0 0xec700000 0 0x10000>;
-			/* placeholder */
+			interrupts =	<GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 320 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 321 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 322 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 323 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 324 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 325 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 327 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 328 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 329 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 331 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 335 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3",
+					  "ch4", "ch5", "ch6", "ch7",
+					  "ch8", "ch9", "ch10", "ch11",
+					  "ch12", "ch13", "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 502>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A774B1_PD_ALWAYS_ON>;
+			resets = <&cpg 502>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
 		};
 
 		audma1: dma-controller@ec720000 {
+			compatible =	"renesas,dmac-r8a774b1",
+					"renesas,rcar-dmac";
 			reg = <0 0xec720000 0 0x10000>;
-			/* placeholder */
+			interrupts =	<GIC_SPI 351 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 337 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 339 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 341 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 343 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 345 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 347 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 349 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 382 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 383 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3",
+					  "ch4", "ch5", "ch6", "ch7",
+					  "ch8", "ch9", "ch10", "ch11",
+					  "ch12", "ch13", "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 501>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A774B1_PD_ALWAYS_ON>;
+			resets = <&cpg 501>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
 		};
 
 		rcar_sound: sound@ec500000 {
-- 
1.9.1

